substrate/frame/staking/src/mock.rs

```diff
@@ -548,10 +548,15 @@ impl ExtBuilder {
 			// having `timestamp::on_initialize` called before `staking::on_initialize`. Also, if
 			// session length is 1, then it is already triggered.
 			ext.execute_with(|| {
-				System::set_block_number(1);
-				Session::on_initialize(1);
-				<Staking as Hooks<u64>>::on_initialize(1);
-				Timestamp::set_timestamp(INIT_TIMESTAMP);
+				System::run_to_block_with::<AllPalletsWithSystem>(
+					1,
+					frame_system::RunToBlockHooks::default().after_initialize(|_| {
+						Timestamp::set_timestamp(INIT_TIMESTAMP);
+					}),
+				);
```

Comment:
Can we not just call the `run_to_block` function here? (I know that the timestamp would be different, but the current setting here is wrong any way)

Reply:
But the origin code appears to be equivalent to running to block 1?

Reply:
What I mean is to use `run_to_block` from this file and not call `System::run_to_block` manually.

Reply:
Okay, then reset the timestamp.

Make sense, to keep logic in one place.